diff -urN dctc-0.62.0.org/src/db.c dctc-0.62.0/src/db.c
--- dctc-0.62.0.org/src/db.c	Thu Dec 20 03:04:05 2001
+++ dctc-0.62.0/src/db.c	Thu Dec 20 03:05:12 2001
@@ -403,6 +403,7 @@
 	
 	G_LOCK(shared_dirs);
 	new_shared=g_array_new(FALSE,FALSE,sizeof(DB_ENTRY));
+	if (shared_dirs != NULL) {
 	for(i=0;i<shared_dirs->len;i++)
 	{
 		int lvl=0;
@@ -410,6 +411,7 @@
 		
 		new_shared=rebuild_dir_database(new_shared,&sod,g_ptr_array_index(shared_dirs,i),&ls_format,lvl);
 	}
+	}
 
 	G_UNLOCK(shared_dirs);
 
