diff -Nru src/Makefile.nomakedepend src-new/Makefile.nomakedepend
--- src/Makefile.nomakedepend	Sat Nov 24 08:57:07 2001
+++ src-new/Makefile.nomakedepend	Sun Dec  2 15:35:42 2001
@@ -1,5 +1,5 @@
-CC = gcc
-CFLAGS =-g -O2 -Wall `glib-config --cflags glib gthread` -Wpointer-arith -Wcast-qual -Wwrite-strings #-Wshadow
+CC = $(gcc)
+CFLAGS =-g -Wall `glib-config --cflags glib gthread` -Wpointer-arith -Wcast-qual -Wwrite-strings $(pldcflags)
 LDFLAGS = -g `glib-config --libs glib gthread`
 
 CSRC = main.c action.c dc_com.c display.c key.c keyboard.c network.c typical_action.c dc_manage.c db.c he3.c gts.c user_manage.c md.c timed_out_string.c sema.c
